tests/Backend/CommonPart1/CreateTableWithConfigurationTest.php

```diff
@@ -0,0 +1,344 @@
+<?php
+
+namespace Keboola\Test\Backend\CommonPart1;
+
+use Keboola\StorageApi\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\Options\Components\Configuration;
+use Keboola\StorageApi\Options\TableWithConfigurationOptions;
+use Keboola\Test\ClientProvider\ClientProvider;
+use Keboola\Test\StorageApiTestCase;
+
+class CreateTableWithConfigurationTest extends StorageApiTestCase
+{
+    public const COMPONENT_ID = 'keboola.app-custom-query-manager';
+
+    private ClientProvider $clientProvider;
+    private Client $client;
+    private Components $componentsClient;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        // check feature
+        $token = $this->_client->verifyToken();
+        if (!in_array('tables-with-configuration', $token['owner']['features'])) {
+            $this->markTestSkipped(sprintf('Creating tables from configurations feature is not enabled for project "%s"', $token['owner']['id']));
+        }
+
+        // init buckets
+        $this->initEmptyTestBucketsForParallelTests();
+
+        $this->clientProvider = new ClientProvider($this);
+        $this->client = $this->clientProvider->createClientForCurrentTest();
+
+        // init configurations
+        try {
+            $this->cleanupConfigurations();
+        } catch (ClientException $e) {
+            if (preg_match('/Configuration cannot be deleted because it is being used in following configured tables: (.*). Delete them first./', $e->getMessage(), $out)) {
+                $tablesToDelete = explode(', ', $out[1]);
+                foreach ($tablesToDelete as $tableId) {
+                    $this->client->dropTable($tableId);
+                }
+            }
+            $this->cleanupConfigurations();
+        }
+
+        // check component exists
+        $this->componentsClient = new Components($this->client);
+        $component = $this->componentsClient->getComponent(self::COMPONENT_ID);
+        $this->assertEquals(self::COMPONENT_ID, $component['id']);
```

Comment:
Musí existovat komponenta? Ono se to někde validuje? Já měl pocit, že ne.

Reply:
no bez komponenty neudelas konfiguraci, je na to check primo v databazi

Reply:
Aha!